src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -1298,12 +1285,16 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		while (currentIndex >= 0 && currentIndex <= this.groupView.count - 1) {
 			currentIndex = anchorIndex < editorIndex ? currentIndex - 1 : currentIndex + 1;
 
+			if (!this.tabsModel.isSelected(currentIndex)) {
+				break;
+			}
+
 			const currentEditor = this.groupView.getEditorByIndex(currentIndex);
-			if (!currentEditor || !this.groupView.isSelected(currentEditor)) {
+			if (!currentEditor) {
 				break;
 			}
 
-			this.groupView.unSelectEditor(currentEditor);
+			this.groupView.unselectEditors([currentEditor]);
```

Comment:
Are we missing an `await` here? Similar for the use of `selectEditors` in this file?